### PR TITLE
Adding 1D Flightdistance tagging variables from cmssw 8_0_0_pre6

### PR DESCRIPTION
**Pull request by IHEP-CMS**
Dear Experts,
                                   An update for the tagging variables has been made after implementing flightdistance1d as well. Please do have a look and merge if the things seems fine.

Cheers,
Sarmad Masood Shaheen.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

@IHEP-CMS 
- please change the PR title/subject to reflect that 1D flight distance tagging variables are added
- in the PR description (the first block that starts now with "Dear experts") please add a brief summary about changes in physics performance, if any; and a link to slides in a POG meeting.

File: DataFormats/BTauReco/interface/TaggingVariable.h [hunk 1]
```diff
@@ -78,7 +78,9 @@ namespace reco {
 
       vertexEnergyRatio,                        // ratio of energy at secondary vertex over total energy
       vertexJetDeltaR,                          // pseudoangular distance between jet axis and secondary vertex direction
-
+      
+      flightDistance1dVal,                      // Linear distance between primary and secondary vertex
```
Comment (ferencek) [inline DataFormats/BTauReco/interface/TaggingVariable.h]: "linear distance" is probably not the best description. Maybe "longitudinal distance" or "longitudinal distance along the z axis"...

**Comment by slava77** [comment]
@ferencek @pvmulder 
Please clarify if this was already presented in the BTV meetings.

**Comment by imarches** [comment]
@slava77 yes, this PR was coordinated with BTV. I will remember to Sarmad to fulfill your request

File: DataFormats/BTauReco/interface/TaggingVariable.h (hunk 1 above)
Comment (IHEP-CMS) [inline DataFormats/BTauReco/interface/TaggingVariable.h]: Thanks for a nice suggestion. The related changes has been made.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
There is a segfault in the unit tests is present in the IB as well.
```
Failure using ttSemiLepSignalSelMVATrainTreeSaver_cfg.py: status 139
```
So, this is not related to this PR.

File: DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ class TemplatedSecondaryVertexTagInfo : public BaseTagInfo {
 
         struct VertexData {
                 VTX                             vertex;
-                Measurement1D                   dist2d, dist3d;
+                Measurement1D                   dist1d,dist2d, dist3d;
                 GlobalVector                    direction;
 		
 		// Used by ROOT storage
```
Comment (slava77) [inline DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h]: I suppose, the class version has to be updated.

@Dr15Jones  do we have a method to require a version change in a templated type after the data members change

**Comment by slava77** [comment]
After fixing the class version locally, it is now possible to read old and new files.

Here is an example of  pfInclusiveSecondaryVertexFinderTagInfos in ttbar (no PU) workflow 25.0 [in these plots red color is for the baseline: values there come out at zero as expected; the new is in the black histogram and shows something sensible)

dist1d.value
all_origvstest13419-pass-bdf83c6_ttbarwf25p0c_recocandidateedmptrsrecojettaginforecoiptaginforecovertexcompositeptrcandidaterecotemplatedsecondaryvertextaginfos_pfinclusivesecondaryvertexfindertaginfos__reco_obj_m_svdata_dist1d_value

dist1d.error
all_origvstest13419-pass-bdf83c6_ttbarwf25p0c_recocandidateedmptrsrecojettaginforecoiptaginforecovertexcompositeptrcandidaterecotemplatedsecondaryvertextaginfos_pfinclusivesecondaryvertexfindertaginfos__reco_obj_m_svdata_dist1d_error

File: DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h (hunk 2 above)
Comment (Dr15Jones) [inline DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h]: The version does need to be updated.

I'm not sure if putting the checksums for each template instantiation into the appropriate classes_def.xml would catch the problem or not.

Comment (IHEP-CMS) [inline DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h]: Dear Salva, 
                        I agree the class version need to be updated after change the template thus may need to include an updated checksum. Please do let me know if I have to update classes_def.xml file as well.

Comment (slava77) [inline DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h]: Hi Sarmad,
Please update the version in this file (from 11 to 12 in two places).
I will need some instructions from Chris on how to update classes_def.xml with checksums included.
If I understand correctly, the change in classes_def.xml is not required at this point.

Comment (IHEP-CMS) [inline DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h]: In classes_def.xml file right?
Also there may be some older version which should not be replaced. I mean I will keep the other version as it is but will add version 12 for Measurement1D. Please do correct me if I am wrong.

Comment (slava77) [inline DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h]: as I noticed in the previous comment, the class version should be updated in this file
(TemplatedSecondaryVertexTagInfo.h)

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/13419	8d054e1
- code changes look ok, in line with the description: dist1d variable is added to the tag infos
- locally run tests pass and comparisons with baseline show differences only in dist1d distributions which are now filled (see plots earlier)


jenkins tests apparently got stuck.
The previous iteration of jenkins tests showed no differences in DQM comparisons; there were false-positive differences in ZEEFS plots using fwlite-based script, but these were affected by inappropriate class version resolution.

**Comment by slava77** [comment]
@cmsbuild please test
it looks like the last attempt got stuck  (https://cmssdt.cern.ch/jenkins/job/ib-any-integration/11512/console the console is showing no activity since somewhere in the middle of the code checkout process)

**Pull request merged**